project_generator/project.py

```diff
@@ -169,68 +170,61 @@ def __init__(self, name, project_files, pgen_workspace):
         """initialise a project with a yaml file"""
         self.project = {}
         self.workspace = pgen_workspace
-
+        self.tool_specific = defaultdict(ToolSpecificSettings)
         self.name = name
-
-        self.include_paths = []
-        self.source_paths = []
-        self.source_groups = {}
-        self.macros = []
-        self.project_dir = {
-            'name': '',
-            'path': '',
-        }
-        self.build_dir = 'build'
-        self.output_dir = {
-            'path': '',
-            'rel_path': '',
-            'rel_count': '',
-        }
-        self.mcu = ''
-        self.core = ''
-        self.target = ''
-        self.tools_supported = []
-
         self.output_types = {
             'executable': 'exe',
             'exe': 'exe',
             'library': 'lib',
             'lib': 'lib',
         }
-        self.output_type = self.output_types['executable']
-
-        self.debugger = 'cmsis-dap'
-
-        self.linker_file = None
-        self.tool_specific = defaultdict(ToolSpecificSettings)
-
-        self.project_name = None
         self.tools = ToolsSupported()
+        self.source_groups = {}
+        self._fill_defaults()
 
         for project_file in project_files:
             try:
                 f = open(project_file, 'rt')
                 project_file_data = yaml.load(f)
-                self.set_attributes(project_file_data)
+                self._set_attributes(project_file_data)
             except IOError:
                raise IOError("The file %s referenced in main yaml doesn't exist."%project_file)
 
-        if self.project_dir['path'] == '':
-            self.project_dir['path'] = self.workspace.settings.generated_projects_dir_default
+    def _fill_defaults(self):
+        self.project['name'] = self.name
```

Comment:
aren't we missing here some keywords? Shouldn't we create a  template for default pgen dictionary? From an external tool, using this, it would like to get that template, fill in own data and use pgen API.

This is an example how the previous pgen dict was used by an external tool, https://github.com/ARMmbed/valinor/blob/master/valinor/main.py#L101